spine-lua/AnimationState.lua

```diff
@@ -720,12 +721,7 @@ function AnimationState:addAnimation (trackIndex, animation, loop, delay)
     if delay <= 0 then
       local duration = last.animationEnd - last.animationStart
       if duration ~= 0 then
-				if last.loop then
-					delay = delay + duration * (1 + math_floor(last.trackTime / duration))
-				else
-					delay = delay + duration
-				end
-        delay = delay - data:getMix(last.animation, animation)
+        delay = delay + duration * (1 + math_floor(last.trackTime / duration)) - data:getMix(last.animation, animation)

```

Comment:
Doesn't this change the logic entirely?

Reply:
I don't know where that update came from... I wasn't trying to change it, I think it might have been an old commit that I cherry-picked. 30 days out and I have no idea what I was thinking :)